Rename service.Application to service.Collector

eddyleelin opened the pull request:
**Description:**
Breaking change: Changed the naming of the collector server to service.Collector (previously named service.Application, which could be confused with the user applications to instrument).

**Link to tracking Issue:** 
Addresses [#3175](https://github.com/open-telemetry/opentelemetry-collector/issues/3175)

**Testing:**
Local `make` does not report any errors in the files changed.

**Documentation:**
- [x] Add number of upstream PR.

**TODO**
- [x] Make changes to contrib repo: [PR 3699](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3699)

carlosalberto commented:
Small nit: _maybe_ `coll` as the common var name is better than `col`, but that's just me (then again, I'm not quite familiar with the go idioms yet ;) )

rakyll commented on service/settings.go:
```diff
@@ -40,8 +40,8 @@ type svcSettings struct {
-// AppSettings holds configuration for creating a new Application.
-type AppSettings struct {
+// ColSettings holds configuration for creating a new Collector.
+type ColSettings struct {
```
Can we call this CollectorSettings? ColSettings is rather harder to associate with the collector because Col is not a common shorten name for collector unlike App is for Appllication.

rakyll reviewed (approved):
One suggestion, otherwise LGTM.

rakyll commented on .github/workflows/snyk-container-analysis.yml:
```diff
@@ -0,0 +1,43 @@
+# A sample workflow which checks out the code, builds a container
```
I don't think this change belongs to this PR, am I right?

rakyll commented on service/collector_windows.go:
```diff
@@ -79,36 +79,36 @@ func (s *WindowsService) Execute(args []string, requests <-chan svc.ChangeReques
 }
 
-func (s *WindowsService) start(elog *eventlog.Log, appErrorChannel chan error) error {
+func (s *WindowsService) start(elog *eventlog.Log, colErrorChannel chan error) error {
```
We probably want to rename the WindowsService to WindowsCollector.

rakyll replied:
Maybe, ignore this for now and we can take another loop if any naming is required before we cut the stable release.

tigrannajaryan commented:
I think there is a commit from @xukaren who did not sign the CLA that's why the check is failing. It is best to avoid having multiple people committing to the same PR to avoid confusion.

rakyll commented:
The commits coming from xukaren is not relevant to this PR. Can we fix the PR? This is a big breaking change and we need to merge it, sooner the better.

xukaren commented:
> I think there is a commit from @xukaren who did not sign the CLA that's why the check is failing. It is best to avoid having multiple people committing to the same PR to avoid confusion.

I think my commits were included by mistake, and I don't see them in the files changed for this PR. 
I've signed the CLA anyway and hopefully this PR is unblocked now.

alolita commented:
@rakyll will get this PR fixed. It might be easier to open a new PR. 

@eddyleelin @xukaren please sync up and fix this issue.

eddyleelin commented:
I apologize for including the incorrect commits by mistake (again, my fault!). 

I have rebased all commits that had previously got approved; they now show up with the new commits 5d1987d..ad55466, so the history for the PR should now be clean (even though the previous commits will still show up on this page). It also includes commit e4fb185 to fixup a small issue where I had deleted a new line on accident during a merge conflict.

Thanks again to everyone for the patience and help along the way!

alolita commented:
@eddyleelin thanks for fixing the commit history. Can you make sure the required tests are passing?

bogdandrutu commented:
@eddyleelin please rebase to fix some of the errors fixed on head

eddyleelin commented:
Awesome, thanks @bogdandrutu !

hankveal181 commented:
Metropolitan Commercial Bank
